tests/AerospikeCacheTest.php

```diff
@@ -3,7 +3,200 @@
+        $aerospikeMock->expects($this->once())
+            ->method('initKey')
+            ->willReturn($aerospikeKey);
+
+        $aerospikeMock->expects($this->once())
+            ->method('get')
+            ->with($aerospikeKey)
+            ->willReturn($statusCode);
+
+        $hasItem = $aerospikeCache->hasItem('foo');
+
+        $this->assertSame($expectedReturnedValue, $hasItem);
+    }
+
+    /**
+     * @dataProvider provideStatusCodes
+     */
+    public function testShouldSaveItem(int $aerospikeStatusCode, bool $expectedSaveSuccessful): void
+    {
+        $aerospikeMock = $this->createMock(\Aerospike::class);
+        $aerospikeCache = new AerospikeCache($aerospikeMock, 'test', 'cache');
+
+        $aerospikeMock->method('put')
+                ->willReturn($aerospikeStatusCode);
+
+        $aerospikeMock->method('get')
+                ->willReturn(\Aerospike::OK);
+
+        $aerospikeKey = ['ns' => 'test', 'set' => 'cache', 'key' => 'foo'];
+        $aerospikeMock->method('initKey')
+                ->willReturn($aerospikeKey);
+
+        $cacheItem = $aerospikeCache->getItem('foo');
+        $saveSuccessful = $aerospikeCache->save($cacheItem);
+
+        $this->assertSame($expectedSaveSuccessful, $saveSuccessful);
+    }
+
+    /**
+     * @dataProvider provideStatusCodes
+     */
+    public function testShouldClearWithEmptyNamespaceName(int $aerospikeStatusCode, bool $expectedClearSuccessful): void
+    {
+        $aerospikeMock = $this->createMock(\Aerospike::class);
+        $aerospikeCache = new AerospikeCache($aerospikeMock, 'test', 'cache');
+
+        $aerospikeMock->method('truncate')
+            ->willReturn($aerospikeStatusCode);
+
+        $clearSuccessful = $aerospikeCache->clear();
+
+        $this->assertSame($expectedClearSuccessful, $clearSuccessful);
+    }
+
+    public function doClearWithEmptyNamespaceProvider(): array
+    {
+        return [[\Aerospike::OK, true] , [\Aerospike::ERR_CLIENT, false]];
+    }
+
+    /**
+     * @dataProvider provideStatusCodesForClearingNamespace
+     */
+    public function testShouldClearNamespace(int $statusCodeForRemove, int $statusCodeForScan, bool $expectedValue): void
+    {
+        $aerospikeMock = $this->createMock(\Aerospike::class);
+        $aerospikeCache = new AerospikeCache($aerospikeMock, 'test', 'cache', 'testNamespace');
+
+        $aerospikeMock->method('remove')
+            ->willReturn($statusCodeForRemove);
+
+        $aerospikeMock->method('scan')
+            ->willReturnCallback(function ($namespace, $set, $callback) use ($statusCodeForScan) {
+                $callback(['key' => ['key' => 'testNamespace::test']]);
+
+                return $statusCodeForScan;
+            });
+
+        $clearResult = $aerospikeCache->clear();
+
+        $this->assertSame($expectedValue, $clearResult);
+    }
+
+    public function provideStatusCodesForClearingNamespace(): array
+    {
+        return [
+            [\Aerospike::OK, \Aerospike::OK, true],
```

Comment:
Some annotation here would be nice (at least).. I'm not sure what those statuses mean..

```php
return [
    // status for ..., status for ..., expected ...
    [OK, OK, true],
    // ..
];
```

Or you can add key to every value, but I'd prefer just a comment.

---
It could be nice to have in other providers as well.

Reply:
(I'd prefer not to add the comment nor key at all, it is unnecessary in my opinion :thinking:  )